speech/cloud-client/beta_snippets.py

```diff
@@ -156,33 +152,42 @@ def transcribe_file_with_diarization(speech_file):
 
     config = speech.types.RecognitionConfig(
         encoding=speech.enums.RecognitionConfig.AudioEncoding.LINEAR16,
-        sample_rate_hertz=16000,
+        sample_rate_hertz=8000,
         language_code='en-US',
         enable_speaker_diarization=True,
         diarization_speaker_count=2)
 
     print('Waiting for operation to complete...')
     response = client.recognize(config, audio)
 
-    for i, result in enumerate(response.results):
-        alternative = result.alternatives[0]
-        print('-' * 20)
-        print('First alternative of result {}: {}'
-              .format(i, alternative.transcript))
-        print('Speaker Tag for the first word: {}'
-              .format(alternative.words[0].speaker_tag))
+    # response.results contains partial results with the last item
+    # containing the entire result:
+    result = response.results[-1]
+
+    words_info = result.alternatives[0].words
+
+    # Separating the words by who said what:
+    speakers_words = []
+    for word_info in words_info:
+        if speakers_words and speakers_words[-1][0] == word_info.speaker_tag:
+            speakers_words[-1][1].append(word_info.word)
+        else:
+            speakers_words.append((word_info.speaker_tag, [word_info.word, ]))
```

Comment:
This is a bit hard to read. An intermediate variable and a `namedtuple` would go a long way to making this clearer:

```py
Speaker = collections.namedtuple('Speaker', ['tag', 'words'])
speaker_words = [Speaker(tag=0, words=[])]
for word_info in words_info:
  current_speaker = speaker_words[-1]
  if current_speaker.tag == word_info.speaker_tag:
    current_speaker.words.append(word_info.word)
  else:
    speaker_words.append(Speaker(tag=word_info.speaker_tag, words=[word_info.word]))
```

Reply:
Also, maybe `speaker_sequence` or something, to make it clear it's not just a speaker->words_they_spoke mapping, and is actually the conversation / words spoken in sequence.

Reply:
Interesting idea. However, I think the code readability is mainly the result of the intermediate`Speaker(tag=0, words=[])` as the first element in the list, allowing us to define `current_speaker` every time.
The downside of it is that we are introducing a new object that is not returned by the API and we will have to handle it in the next step separately (either by removing it, or by skipping it), which reduces the code cleanness in another way.
So, while I am not against the suggested solution, I am also not sure if it is really helping with the code readability that much.
What do you think?

Reply:
What about this:

```
    speakers = []
    words = []
    for word_info in words_info:
        if speakers and speakers[-1] == word_info.speaker_tag:
            words[-1].append(word_info.word)
        else:
            speakers.append(word_info.speaker_tag)
            words.append([word_info.word, ])
```
I think this is more readable that the current code, without introducing the intermediate variable.